Momento/ScsDataClient.cs

```diff
@@ -59,24 +60,102 @@ public async Task<CacheSetResponse> SetAsync(string cacheName, string key, strin
 
         public async Task<CacheGetResponse> GetAsync(string cacheName, string key)
         {
-            _GetResponse resp = await this.SendGetAsync(cacheName, Convert(key), dataClientOperationTimeoutSeconds: this.dataClientOperationTimeoutSeconds);
+            _GetResponse resp = await this.SendGetAsync(cacheName, Convert(key), dataClientOperationTimeoutMilliseconds: this.dataClientOperationTimeoutMilliseconds);
             return new CacheGetResponse(resp);
         }
 
         public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[] value, uint ttlSeconds)
         {
-            _SetResponse response = await this.SendSetAsync(cacheName, value: Convert(value), key: Convert(key), ttlSeconds: ttlSeconds, dataClientOperationTimeoutSeconds: this.dataClientOperationTimeoutSeconds);
+            _SetResponse response = await this.SendSetAsync(cacheName, value: Convert(value), key: Convert(key), ttlSeconds: ttlSeconds, dataClientOperationTimeoutMilliseconds: this.dataClientOperationTimeoutMilliseconds);
             return new CacheSetResponse(response);
         }
 
-         public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[] value)
+        public async Task<CacheSetResponse> SetAsync(string cacheName, string key, byte[] value)
         {
             return await this.SetAsync(cacheName, key, value, defaultTtlSeconds);
         }
 
+        public async Task<CacheMultiGetResponse> MultiGetAsync(string cacheName, List<string> keys)
+        {
+            List<Task<CacheMultiGetResponse>> tasks = new();
+            List<CacheGetResponse> successResponses = new();
+            List<CacheMultiGetFailureResponse> failedResponses = new();
+            foreach (string key in keys)
+            {
+                tasks.Add(SendMultiGetAsync(cacheName, Convert(key)));
+            }
+
+            await Task.WhenAll(tasks);
+            foreach (Task<CacheMultiGetResponse> t in tasks)
+            {
+                if (t.Result.SuccessfulResponse() is not null)
+                {
+                    successResponses.Add(t.Result.SuccessfulResponse());
+                }
+                if (t.Result.FailedResponse() is not null)
+                {
+                    failedResponses.Add(t.Result.FailedResponse());
+                }
+            }
+
+            return new CacheMultiGetResponse(successResponses, failedResponses);
```

Comment:
nit: feel like this is a bit much code repetition. would be nice to clean this up a bit. just a nit.